src/phantom/schema.py

```diff
@@ -44,6 +47,9 @@ def __schema__(cls) -> Schema:
         for more information. This hook differs to pydantic's ``__modify_schema__()``
         and expects subclasses to instantiate new dicts instead of mutating a given one.
 
+        Note that the docstring attached to a phantom type class, if present,
+        will override any description that is defined in `__schema__()`.
```

Comment:
I don't think this is the behavior we want to introduce. We should still allow users to override with an explicit value for `"description"` in `__schema__`. A really easy way to achieve that should be to move the implementation from `__modify_schema__` to `__schema__` where you can literally just do `{..., "description": cls.__doc__}`.

Reply:
But I don't think that it will have the intended effect. Consider:

```python
class FancyScore(float, Closed, low=0, high=10):
    """My fancy description"""
```

Now if I call `schema()` on a model using it, as I understand, the chain would go:

* `FancyScore.__modify_schema__()` (which is `SchemaField.__modify_schema__`)
* uses `FancyScore.__schema__()`
* which inherits `Closed.__schema__()` **<- provides a default description overriding what follows below**
* which uses `super().__schema__()` (possibly through a few superclasses)
* which uses `SchemaField.__schema__()` **<- where you propose to add the docstring**

If you think that `__schema__` should have priority over the docstring, then a different approach is needed - in that case we need to recognize where the `description` came from - was it overridden in the same class inside `__schema__`, or was it inherited from a base class? If it was inherited, the docstring of the current class should provide the default.

Without changing changing the general machinery, this would only be achievable ad-hoc by modifying all `__schema__` methods. Actually, not sure how this could be done most elegantly without sandwiching some other method into the cascade or introducing metaclass magic (like: check `__schema__` of the base class and the new class, compare `description` - if it changed, then it was overriden, otherwise it was inherited. If it was not overridden, use docstring as description)

Reply:
Ah, thanks for the detailed description of the problem, I didn't have all the pieces in my head. I'll have to tinker with it a bit to see if I can come up with something. Otherwise I'll probably be reluctant to introduce this feature, unfortunately.

Reply:
Sure, no problem! I think the Meta class approach could work, in principle. 

When a new class is subclassed:  
* no docstring -> all good
* docstring, no defined schema method -> generate schema method with docstring as description 
* docstring, defines schema method -> call both inherited and new method to see if description (if any) is "new", if not there or not new, wrap method to add the docstring to the result. 

That way it would have the desired behavior, but it's up to you to decide if this is an acceptable solution. The last case is "ugly", but still it's a check needed only once during class creation, and it could (I actually expect it to) be a rare case that both things are present.

Reply:
So your modification (over what I proposed) is to ask the user (by requiring the flag) to confirm explicitly that the docstring should override the schema description, right?

I would be fine with that, especially if it's called `use_docstring` - it would still reduce a lot of boilerplate :)